web_src/js/features/comp/Cropper.ts

```diff
@@ -1,47 +1,67 @@
-import {showElem, type DOMEvent} from '../../utils/dom.ts';
+import {createElementFromHTML, hideElem, showElem, type DOMEvent} from '../../utils/dom.ts';
+import {debounce} from 'perfect-debounce';
+import type {CropperCanvas, CropperSelection} from 'cropperjs';
 
 type CropperOpts = {
   container: HTMLElement,
-  imageSource: HTMLImageElement,
+  wrapper: HTMLDivElement,
   fileInput: HTMLInputElement,
 }
 
-async function initCompCropper({container, fileInput, imageSource}: CropperOpts) {
-  const {default: Cropper} = await import(/* webpackChunkName: "cropperjs" */'cropperjs');
-  let currentFileName = '';
-  let currentFileLastModified = 0;
-  const cropper = new Cropper(imageSource, {
-    aspectRatio: 1,
-    viewMode: 2,
-    autoCrop: false,
-    crop() {
-      const canvas = cropper.getCroppedCanvas();
+async function initCompCropper({container, fileInput, wrapper}: CropperOpts) {
+  await import(/* webpackChunkName: "cropperjs" */'cropperjs');
+
+  fileInput.addEventListener('input', (e: DOMEvent<Event, HTMLInputElement>) => {
+    if (!e.target.files?.length) {
+      wrapper.replaceChildren();
+      hideElem(container);
+      return;
+    }
+
+    const [file] = e.target.files;
+    const objectUrl = URL.createObjectURL(file);
+    const cropperCanvas = createElementFromHTML<CropperCanvas>(`
+      <cropper-canvas theme-color="var(--color-primary)">
+        <cropper-image src="${objectUrl}" scalable skewable translatable></cropper-image>
+        <cropper-shade hidden></cropper-shade>
+        <cropper-handle action="select" plain></cropper-handle>
+        <cropper-selection aspect-ratio="1" movable resizable>
+          <cropper-handle action="move" theme-color="transparent"></cropper-handle>
+          <cropper-handle action="n-resize"></cropper-handle>
+          <cropper-handle action="e-resize"></cropper-handle>
+          <cropper-handle action="s-resize"></cropper-handle>
+          <cropper-handle action="w-resize"></cropper-handle>
+          <cropper-handle action="ne-resize"></cropper-handle>
+          <cropper-handle action="nw-resize"></cropper-handle>
+          <cropper-handle action="se-resize"></cropper-handle>
+          <cropper-handle action="sw-resize"></cropper-handle>
+        </cropper-selection>
+      </cropper-canvas>
+    `);
+    cropperCanvas.querySelector<CropperSelection>('cropper-selection').addEventListener('change', debounce(async (e) => {
```

Comment:
This event is not right. For example: "zoom" won't trigger "change" event and users will get wrong image.

Reply:
Ideally I would like to disable zooming, but found no way to do so.